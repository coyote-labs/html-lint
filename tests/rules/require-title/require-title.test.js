const {htmlLint} = require('../../../index');

global.console = {
  error: jest.fn()
};

test.each([
  'tests/rules/require-title/no-error.html',
  'tests/rules/require-title/error-no-title.html',
  'tests/rules/require-title/error-no-head.html'
])(
  'require-title %s',
  async(fixture) => {
    try {
      await htmlLint(fixture);
    } catch ({ message }) {
      expect(global.console.error).toMatchSnapshot();
    }
  },
);
