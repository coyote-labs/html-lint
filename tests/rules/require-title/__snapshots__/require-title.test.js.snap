// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`require-title tests/rules/require-title/error-no-head.html 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "[101m[html-lint] 1 error in tests/rules/require-title/error-no-title.html[49m",
    ],
    Array [
      "[90mrequire-title: <title> must be present in the <head> tag[39m
[0m [90m 1 | [39m[33m<[39m[33m![39m[33mDOCTYPE[39m html[33m>[39m[0m
[0m [90m 2 | [39m[33m<[39m[33mhtml[39m[33m>[39m[0m
[0m[31m[1m>[22m[39m[90m 3 | [39m  [33m<[39m[33mheaD[39m[33m>[39m[0m
[0m [90m   | [39m  [31m[1m^[22m[39m [31m[1m<title> must be present in the <head> tag[22m[39m[0m
[0m [90m 4 | [39m  [33m<[39m[33m/[39m[33mheaD[39m[33m>[39m[0m
[0m [90m 5 | [39m[33m<[39m[33m/[39m[33mhtml[39m[33m>[39m[0m",
      "
",
    ],
    Array [
      "
",
    ],
    Array [
      "[101m[html-lint] 1 error in tests/rules/require-title/error-no-head.html[49m",
    ],
    Array [
      "[90mrequire-title: <title> must be present in the <head> tag[39m
[0m [90m 1 | [39m[33m<[39m[33m![39m[33mDOCTYPE[39m html[33m>[39m[0m
[0m[31m[1m>[22m[39m[90m 2 | [39m[33m<[39m[33mhtml[39m[33m>[39m[0m
[0m [90m   | [39m[31m[1m^[22m[39m [31m[1m<title> must be present in the <head> tag[22m[39m[0m
[0m [90m 3 | [39m  [33m<[39m[33mbody[39m[33m>[39m[0m
[0m [90m 4 | [39m    [33m<[39m[33mp[39m[33m>[39m[33mThis[39m is a paragraph[33m<[39m[33m/[39m[33mp[39m[33m>[39m[0m
[0m [90m 5 | [39m  [33m<[39m[33m/[39m[33mbody[39m[33m>[39m[0m",
      "
",
    ],
    Array [
      "
",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;

exports[`require-title tests/rules/require-title/error-no-title.html 1`] = `
[MockFunction] {
  "calls": Array [
    Array [
      "[101m[html-lint] 1 error in tests/rules/require-title/error-no-title.html[49m",
    ],
    Array [
      "[90mrequire-title: <title> must be present in the <head> tag[39m
[0m [90m 1 | [39m[33m<[39m[33m![39m[33mDOCTYPE[39m html[33m>[39m[0m
[0m [90m 2 | [39m[33m<[39m[33mhtml[39m[33m>[39m[0m
[0m[31m[1m>[22m[39m[90m 3 | [39m  [33m<[39m[33mheaD[39m[33m>[39m[0m
[0m [90m   | [39m  [31m[1m^[22m[39m [31m[1m<title> must be present in the <head> tag[22m[39m[0m
[0m [90m 4 | [39m  [33m<[39m[33m/[39m[33mheaD[39m[33m>[39m[0m
[0m [90m 5 | [39m[33m<[39m[33m/[39m[33mhtml[39m[33m>[39m[0m",
      "
",
    ],
    Array [
      "
",
    ],
  ],
  "results": Array [
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
    Object {
      "type": "return",
      "value": undefined,
    },
  ],
}
`;
